src/mrpro/operators/_FastFourierOp.py

```diff
@@ -3,57 +3,112 @@
+            as these correspond to k2, k1, and k0 of KData.
+        encoding_matrix
+            shape of encoded k-data along the axes in dim. Must be set if recon_matrix is set.
+            If encoding_matrix and recon_matrix are None, no padding or cropping will be performed.
+            If all values in dim are -3, -2 or -1, this can also be a SpatialDimension describing the
+            k-space shape in all 3 dimensions (k2, k1, k0), but only values in the dimensions in dim will be used.
+            Otherwise, it should be a Sequence of the same length as dim.
+        recon_matrix
+            shape of reconstructed image data. Must be set if encoding_matrix is set.
+            If encoding_matrix and recon_matrix are None, no padding or cropping will be performed.
+            If all values in dim are -3, -2 or -1, this can also be a SpatialDimension describing the
+            image-space shape in all 3 dimensions (z,y,x), but only values in the dimensions in dim will be used.
+            Otherwise, it should be a Sequence of the same length as dim.
         """
         super().__init__()
         self._dim = tuple(dim)
         self._pad_op: ZeroPadOp
-        if encoding_shape is not None and recon_shape is not None:
-            self._pad_op = ZeroPadOp(dim=dim, original_shape=recon_shape, padded_shape=encoding_shape)
+
+        if isinstance(recon_matrix, SpatialDimension):
+            original_shape: Sequence[int] | None = [int(astuple(recon_matrix)[d]) for d in dim]
+            if not all(d in (-1, -2, -3) for d in dim):
+                raise NotImplementedError(
+                    f'recon_matrix can only be a SpatialDimension if each value in dim is in (-3,-2,-1),'
+                    f'got {dim=}\nInstead, you can also supply a list of values of same length as dim'
+                )
+        else:
+            original_shape = recon_matrix
+
+        if isinstance(encoding_matrix, SpatialDimension):
+            padded_shape: Sequence[int] | None = [int(astuple(encoding_matrix)[d]) for d in dim]
+            if not all(d in (-1, -2, -3) for d in dim):
+                raise NotImplementedError(
+                    f'encoding_matrix can only be a SpatialDimension if each value in dim is in (-3,-2,-1),'
+                    f'got {dim=}\nInstead, you can also supply a list of values of same length as dim'
+                )
         else:
-            # No padding
+            padded_shape = encoding_matrix
+
+        if original_shape is not None and padded_shape is not None:
+            # perform padding / cropping
+            self._pad_op = ZeroPadOp(dim=dim, original_shape=original_shape, padded_shape=padded_shape)
+        elif encoding_matrix is None and recon_matrix is None:
+            # No padding no padding / cropping
             self._pad_op = ZeroPadOp(dim=(), original_shape=(), padded_shape=())
+        else:
```

Comment:
this is new and afaik required. there is currently no good way to do it with only one of the matrices set.
before, it failed in zeropadop